src/json_schema/validators/properties.rs

```diff
@@ -21,16 +20,36 @@ pub struct Properties {
 
 impl super::Validator for Properties {
     fn validate(&self, val: &Value, path: &str, scope: &scope::Scope) -> super::ValidationState {
-        let object = nonstrict_process!(val.as_object(), path);
+        let mut object = Cow::Borrowed(nonstrict_process!(val.as_object(), path));
         let mut state = super::ValidationState::new();
 
-        'main: for (key, value) in object.iter() {
+        if scope.supply_defaults {
+            for (key, url) in self.properties.iter() {
+                if let Some(schema) = scope.resolve(url) {
+                    if object.get(key).is_none() && schema.has_default() {
+                        object
+                            .to_mut()
+                            .insert(key.clone(), schema.get_default().unwrap());
+                    }
+                }
+            }
+        }
+
+        // necessary due to object being mutated in the loop
+        let keys = object.keys().cloned().collect::<Vec<_>>();
+        'main: for key in keys.iter() {
```

Comment:
does that need to be named? you don't seem to break anyway

Reply:
I left it in because it clarifies the `continue` that appears many lines down.